util/src_util/ut_tc_apply.c

```diff
@@ -121,7 +121,7 @@ UTEST(TC_APPLY_SECURITY, HAPPY_PATH_ENC)
 
     int32_t return_val = CRYPTO_LIB_ERROR;
 
-    SecurityAssociation_t* test_association = malloc(sizeof(SecurityAssociation_t) * sizeof(uint8_t));
+    SecurityAssociation_t* test_association;
```

Comment:
Should we locally allocating a full up `SecurityAssociation_t` then dereferencing it when passing it?

Reply:
This is just a unit test., but it is possible that could be done.  Worth the effort for unit tests?  This is just grabbing the already allocated test_association, so that a change can be made for this test(s)

Reply:
I understand what you mean now.  I'll fix that in the morning.

Reply:
Going to leave this as is for now, after all.  Changing would require refactoring of the sadb_routine, as it accepts a double pointer.